terraform/modules/open_id_connect/roles.tf

```diff
@@ -0,0 +1,15 @@
+locals {
+  is_development = var.environment == "test"
```

Comment:
Moving the roles into a separate folder from the OIDC provider would probably be easier!

If you want to keep it as is, we try and avoid changing behaviour based on the environment string because it will cause issues if the names of the environments ever change, e.g. a new pre-prod environment which also wouldn't want these roles. Could make this a variable with a more descriptive name e.g. `enable_terraform_roles_for_development_environments`

Reply:
happy to move it into a separate folder if it means cleaner code but unsure how this would work. Do you mean the roles folder would be its own module that I would reference within the main.tf file of whichever environment needed the roles creating?

Reply:
I mean have an (e.g.) "github_actions_openid_connect_provider" module with just the `aws_iam_openid_connect_provider` resource, and a separate "github_actions_dev_terraform_roles" module with what's currently in `policies.tf` and `roles.tf`. Then the first module would be used in test and production, and the second only in test.

Hopefully that will make it clear what's going on from a look through just the environment main.tf files rather than needing to dig into the module.

(and sorry again, I know I'm being very nitpicky about this, but it's security critical code so I want it to be as clear as possible)

Reply:
no worries - thanks for clarifying. Have moved stuff around as suggested